src/lib/y2network/interface.rb

```diff
@@ -35,11 +93,36 @@ def initialize(name)
     # @return [Boolean]
     def ==(other)
       return false unless other.is_a?(Interface)
+      # FIXME: take renaming into account too
       name == other.name
     end
 
+    def type
+      Yast::NetworkInterfaces.GetType(name)
+    end
+
+    def config
+      system_config(name)
+    end
+
+    # Updates object status according to current system state
+    def reload
+      init(@name)
+    end
+
     # eql? (hash key equality) should alias ==, see also
     # https://ruby-doc.org/core-2.3.3/Object.html#method-i-eql-3F
     alias_method :eql?, :==
+
+  private
+
+    def system_config(name)
+      Yast::NetworkInterfaces.devmap(name)
+    end
+
+    # Initializates depending variables
+    def init(name)
```

Comment:
this is strange construct, as you basically have two initialize methods. Either add it directly to initialize or rename it like `read_configured` so it is clear what it does.

Reply:
this has already changed several times. Need to polish once it is stable. Taking note.